### DISCUSSION TITLE
Read YAML for variables/parameters

### DISCUSSION THREAD

**Discussion by tf13** [ttscoff/bunch-docs#113] [Q&A] [answered]
Is it possible for a bunch to read a YAML file for parameters / to fill in variables for subsequent actions? I have a folder of yaml files that contain project details (project directory, links to cloud documents, etc.) and would love to have a single "open project" bunch that draws on those details. 

Happy to use a shell script (I know Python well, and bash, AppleScript and lua slightly) or some other intermediary (hammerspoon, Keybord Maestro both installed).

**Comment by ttscoff** [comment]
See the 'from file:' frontmatter key.

- Brett

**Comment by ttscoff** [comment]
Oh, but I assume you want a menu to select which filter is read? That gets a little trickier, but you could roll your own with an AppleScript dialog called with the 'from script:' frontmatter key. As long as it ultimately returns Bunch's YAML-ish format.

- Brett

**Reply by tf13** [reply]
Ok, cool. Thanks much! Will check that out and see if I can figure it out. If not I'll post back here!

**Reply by tf13** [reply]
Hm. Not sure this works. I have a yaml file for each project, and then a list of project names created from the yaml filenames (by Hazel). As I add projects, Hazel adds to the project list. I'd love to be able to pull in that project list (just a text file, one project name per line) and be able to choose from those. That would then tell bunch which project file to use to populate various variables. I may need to use Keyboard Maestro or something similar to choose the project name, then pass that to Bunch...

**Reply by tf13** [reply]
I guess my Bunch wishlist would include the ability to read from a file to populate an array query or dictionary query.

**Comment by ttscoff** [comment, marked as answer]
Oh, you can totally do that. Write the file in a Bunch query dialog 
format (array or dictionary), then use a shell script to populate a 
variable with its contents:

myvar = $ cat mydialog.txt

Then use the raw transform to output it as part of the Bunch:

${myvar/raw}

You can also use the result to populate a variable with:

newvar = ${myvar/raw}

-Brett
